src/Semantics.js

```diff
@@ -369,8 +370,9 @@ function newDefaultAction(type, name, doIt) {
     } else {
       // Otherwise, we throw an exception to let the programmer know that we don't know what
       // to do with this node.
-      throw new Error(
-          'Missing semantic action for ' + this.ctorName + ' in ' + name + ' ' + type);
+
+      throw errors.semanticErrors(
+          this.ctorName, name, type) ;
```

Comment:
I think it would be simpler to just use a regular return value here, rather than throwing an exception. Exceptions are appropriate for doing a kind of "non-local return", but in this case I think it would be simpler to just return an error value to the caller. Also, try/catch is a known optimization killer in JavaScript, so we should avoiding putting it in hot paths like this.

However, we need to return a value that can't be mistaken for something that could be returned by the programmer's semantic actions. Here's what I'm thinking:

- in `createError`, we could add a field to every error object, that lets us recognize that it's an error created by Ohm. E.g.:

```
var OHM_ERROR = {};

function createError() {
  ...
  e.type = OHM_ERROR;
  return e;
}
```

And then, down below where we do `thisThing.execute`, we could check if `ans` is an Ohm error:

```
if (ans.type === errors.OHM_ERROR) {
  ...
}
```

Make sense?